plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -241,25 +240,36 @@ rules recorder plugin = do
 
       diagnostics :: NormalizedFilePath -> Either ParseError [Idea] -> [FileDiagnostic]
       diagnostics file (Right ideas) =
-        [(file, ShowDiag, ideaToDiagnostic i) | i <- ideas, ideaSeverity i /= Ignore]
+       (file, ShowDiag,) <$> catMaybes [ideaToDiagnostic i | i <- ideas]
       diagnostics file (Left parseErr) =
         [(file, ShowDiag, parseErrorToDiagnostic parseErr)]
 
-      ideaToDiagnostic :: Idea -> Diagnostic
-      ideaToDiagnostic idea =
-        LSP.Diagnostic {
-            _range    = srcSpanToRange $ ideaSpan idea
-          , _severity = Just LSP.DiagnosticSeverity_Information
-          -- we are encoding the fact that idea has refactorings in diagnostic code
-          , _code     = Just (InR $ T.pack $ codePre ++ ideaHint idea)
-          , _source   = Just "hlint"
-          , _message  = idea2Message idea
-          , _relatedInformation = Nothing
-          , _tags     = Nothing
-          , _codeDescription = Nothing
-          , _data_ = Nothing
-        }
-        where codePre = if null $ ideaRefactoring idea then "" else "refact:"
+
+      ideaToDiagnostic :: Idea -> Maybe Diagnostic
+      ideaToDiagnostic idea = do
+        diagnosticSeverity <- ideaSeverityToDiagnosticSeverity (ideaSeverity idea)
+        pure $
+            LSP.Diagnostic {
+              _range    = srcSpanToRange $ ideaSpan idea
+            , _severity = Just diagnosticSeverity
+            -- we are encoding the fact that idea has refactorings in diagnostic code
+            , _code     = Just (InR $ T.pack $ codePre ++ ideaHint idea)
+            , _source   = Just "hlint"
+            , _message  = idea2Message idea
+            , _relatedInformation = Nothing
+            , _tags     = Nothing
+            , _codeDescription = Nothing
+            , _data_ = Nothing
+            }
+
+        where
+          codePre = if null $ ideaRefactoring idea then "" else "refact:"
+
+          ideaSeverityToDiagnosticSeverity :: Hlint.Severity -> Maybe LSP.DiagnosticSeverity
```

Comment:
Could you explain what we are doing here and link to the issue where we discussed it?

Reply:
Done!. Let me know if the comment is detailed enough

Reply:
Great, thanks!